### PR TITLE
[MODULAR] Arousal system and interactions

### PR DESCRIPTION
**Pull request by Lamella-0587**
This pull request containing Arousal system and interactions ported from sandstorm stations. This is the base for our huge ERP update, in the future these files would be changed to work with other .dm files we are going to add. 

I tried to fix all the bugs. Visual and even those that don't affect anything but spam in console, still it possibly can be buggy, please run some tests before merging.

P.S
We have a lot of work to do, but if you want us to finish our update faster - you can support our team on patreon.
https://www.patreon.com/lamella

### CODE REVIEW

**Comment by Swiftfeather** [comment]
Please atomize this. We'd like a functioning interaction menu first, that includes all the interactions already in game, then new interactions can be added later along with an arousal system.

**Comment by Cheshify** [comment]
Worried over the git getting removed, TOS and all. Interaction menu is fine, I think keeping Arousal systems separate and worked on after interaction menu is complete. Really, what Swift said.

**Comment by hugmeorelse** [comment]
Yeah, this isn't something we can have on Github. I'm the host. DM me.

**Comment by gonenoculer5** [comment]
@Lamella-0587 If you wanna fix something relating to this, cumshots with the *cum emote are broken for carbons. And they also dont work for borgs!

**Comment by tf-4** [comment]
> 
> 
> @Lamella-0587 If you wanna fix something relating to this, cumshots with the *cum emote are broken for carbons. And they also dont work for borgs!

This isn't relating to this, and you should make an issue report for bugs

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/datum/interaction/handshake
+	command = "handshake"
+	description = "Shake their hand."
+	simple_message = "USER shakes the hand of TARGET."
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:5]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	needs_physical_contact = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:6]:
```suggestion
	needs_physical_contact = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+/datum/interaction/pat
+	command = "pat"
+	description = "Pat their shoulder."
+	simple_message = "USER pats TARGET's shoulder."
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:12]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	needs_physical_contact = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:13]:
```suggestion
	needs_physical_contact = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+/datum/interaction/cheer
+	command = "cheer"
+	description = "Cheer them on."
+	require_user_mouth = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:18]:
```suggestion
	require_user_mouth = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER cheers TARGET on!"
+
+/datum/interaction/highfive
+	command = "highfive"
+	description = "Give them a high-five."
+	require_user_mouth = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:24]:
```suggestion
	require_user_mouth = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER high fives TARGET!"
+	interaction_sound = 'modular_skyrat/modules/modular_items/lewd_items/sounds/slap.ogg'
+	needs_physical_contact = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:27]:
```suggestion
	needs_physical_contact = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+/datum/interaction/headpat
+	command = "headpat"
+	description = "Pat their head. Aww..."
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:32]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER headpats TARGET!"
+	needs_physical_contact = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:34]:
```suggestion
	needs_physical_contact = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+/datum/interaction/salute
+	command = "salute"
+	description = "Give them a firm salute!"
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:39]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER salutes TARGET sharply!"
+	max_distance = 25
+
+/datum/interaction/fistbump
+	command = "fistbump"
+	description = "Bump it!"
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:46]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER fistbumps TARGET! Yeah!"
+	needs_physical_contact = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:48]:
```suggestion
	needs_physical_contact = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+/datum/interaction/pinkypromise
+	command = "pinkypromise"
+	description = "Make a pinky promise with them!"
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:53]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER hooks their pinky with TARGET's! Pinky Promise!"
+	needs_physical_contact = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:55]:
```suggestion
	needs_physical_contact = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+/datum/interaction/bird
+	command = "bird"
+	description = "Flip them the bird!"
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:60]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER gives TARGET the bird!"
+	max_distance = 25
+
+/datum/interaction/holdhand
+	command = "holdhand"
+	description = "Hold their hand."
+	require_user_hands = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:67]:
```suggestion
	require_user_hands = TRUE
```

File: modular_skyrat/modules/interactions/interaction_definitions.dm [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+	simple_message = "USER holds TARGET's hand."
+	max_distance = 25
+	needs_physical_contact = 1
```
Comment (KathrinBailey) [inline modular_skyrat/modules/interactions/interaction_definitions.dm:70]:
```suggestion
	needs_physical_contact = TRUE
```

**Review by KathrinBailey — COMMENTED** [review]
TG uses defines for true/false instead of 1s and 0s in code for readability's sake, we're trying to follow their standards, batch these in `files changed` and you can commit them all in one.
It would've been easier to edit the file but I didn't wanna do that without asking.

Useroth is gonna want you to do some other stuff too, but this is just, um, TG coding standards stuff.

**Comment by NullDagaf** [comment]
> Worried over the git getting removed, TOS and all. Interaction menu is fine, I think keeping Arousal systems separate and worked on after interaction menu is complete. Really, what Swift said.

I'm not gonna lie, it's kinda confusing that we can have a "cum" emote, but this is too much.

**Comment by RatFromTheJungle** [comment]
> 
> 
> > Worried over the git getting removed, TOS and all. Interaction menu is fine, I think keeping Arousal systems separate and worked on after interaction menu is complete. Really, what Swift said.
> 
> I'm not gonna lie, it's kinda confusing that we can have a "cum" emote, but this is too much.

its the graphic description of "sex" acts, as in the acts themselves that are against T-O-S

**Comment by NopemanMcHalt** [comment]
> > > Worried over the git getting removed, TOS and all. Interaction menu is fine, I think keeping Arousal systems separate and worked on after interaction menu is complete. Really, what Swift said.
> > 
> > 
> > I'm not gonna lie, it's kinda confusing that we can have a "cum" emote, but this is too much.
> 
> its the graphic description of "sex" acts, as in the acts themselves that are against T-O-S

Please explain then, why there's countless (text-only) game repos full of these on github, and yet none of them got taken down?

**Comment by RatFromTheJungle** [comment]
> 
> 
> > > > Worried over the git getting removed, TOS and all. Interaction menu is fine, I think keeping Arousal systems separate and worked on after interaction menu is complete. Really, what Swift said.
> > > 
> > > 
> > > I'm not gonna lie, it's kinda confusing that we can have a "cum" emote, but this is too much.
> > 
> > 
> > its the graphic description of "sex" acts, as in the acts themselves that are against T-O-S
> 
> Please explain then, why there's countless (text-only) game repos full of these on github, and yet none of them got taken down?

not my job, go ask github support 
image
https://docs.github.com/en/github/site-policy/github-community-guidelines

**Comment by Gandalf2k15** [comment]
This is awesome and all, but I'd prefer if you colllaborated more with Useroth and then PR'd this in a safe way. My closing this is not a denial, but I can see this will take some time. So it's best if you get it sorted first 👍

**Pull request closed without merging**